pkg/coscheduling/coscheduling.go

```diff
@@ -233,10 +232,17 @@ func (cs *Coscheduling) calculateTotalPods(podGroupName, namespace string) int {
 	return len(pods)
 }
 
-func (cs *Coscheduling) calculateRunningPods(podGroupName, namespace string) int {
+func (cs *Coscheduling) calculateBindedPods(podGroupName, namespace string) int {
 	pods, err := cs.frameworkHandle.SnapshotSharedLister().Pods().FilteredList(func(pod *v1.Pod) bool {
-		if pod.Labels[PodGroupName] == podGroupName && pod.Namespace == namespace && pod.Status.Phase == v1.PodRunning {
-			return true
+		if pod.Labels[PodGroupName] == podGroupName && pod.Namespace == namespace {
+			if pod.Status.Phase == v1.PodRunning {
+				return true
+			}
+			for _, c := range pod.Status.Conditions {
+				if c.Type == v1.PodScheduled || c.Type == v1.PodInitialized || c.Type == v1.ContainersReady || c.Type == v1.PodReady {
+					return true
+				}
+			}
```

Comment:
Why this is needed?

Reply:
- running := cs.calculateRunningPods(podGroupName, namespace)
It only calculate the pods has bound to a node and all of the containers have been started, which may miss these bound pod whose PodPhase is still in Pending before containers start during pull images.

Is my opinion correct？

Reply:
That's correct - running means at least one of the containers is already running.

Given that the core scheduler doesn't really look at the phase of a Pod, @denkensk do you think we should just check a Pod's spec.nodeName here?

Reply:
After a deep dive into the scheduler framework, I also believe that check spec.nodeName is much better than the Status。Conditions

Reply:
Thanks @Huang-Wei  I agree to just check a Pod's nodeName.

Reply:
Thanks @DeliangFan
If we just check the Pod's `spec.nodeName`,  we need to rename `calculateRunningPods `  to `calculateBindedPods`.

Reply:
Done, please review again